lib/modules/manager/hermit/artifacts.ts

```diff
@@ -0,0 +1,259 @@
+import fs from 'fs-extra';
+import pMap from 'p-map';
+import upath from 'upath';
+import { GlobalConfig } from '../../../config/global';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import type {
+  HermitPackageDependency,
+  ReadContentResult,
+  UpdateHermitError,
+  UpdateHermitResult,
+} from './types';
+
+/**
+ * updateArtifacts runs hermit install for each updated dependencies
+ */
+export async function updateArtifacts({
+  updatedDeps,
+  packageFileName,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug({ packageFileName }, `hermit.updateArtifacts()`);
+  const dependencies = updatedDeps.map(
+    (dep) =>
+      ({
+        ...dep,
+        packageFileName,
+      } as HermitPackageDependency)
+  );
+  try {
+    // maps each updated dependencies to the actual hermit install command,
+    // and runs them one by one.
+    await pMap(dependencies, updateHermitPackage, { concurrency: 1 });
+  } catch (err) {
+    const execErr: UpdateHermitError<any> = err;
+    logger.warn(
+      { stdout: execErr.stdout, stderr: execErr.stderr },
+      `error updating hermit packages.`
+    );
+    return [execErr.from, execErr.to].map((lockFile) => ({
+      artifactError: {
+        lockFile,
+        stderr: execErr.stderr,
+      },
+    }));
+  }
+
+  logger.debug(`scanning the changes after update`);
+
+  const updateResult = await getUpdateResult(packageFileName);
+
+  logger.debug({ updateResult }, `update result for hermit`);
+
+  return updateResult;
+}
+
+/**
+ * getContent returns the content of either link or a normal file
+ */
+async function getContent(file: string): Promise<ReadContentResult> {
```

Comment:
we usually use readLocalFile to read file contents, but we dont support symbolic links,
@rarkins @viceice  should we add this file reader to our `utils/fs` ?

Reply:
I've moved related functions into `util/fs`